Sources/OpenAPIRuntime/EventStreams/ServerSentEventsDecoding.swift

```diff
@@ -46,7 +56,22 @@ extension ServerSentEventsDeserializationSequence: AsyncSequence {
         var upstream: UpstreamIterator
 
         /// The state machine of the iterator.
-        var stateMachine: StateMachine = .init()
+        var stateMachine: StateMachine
+
+        /// A closure that determines whether the given byte chunk should be forwarded to the consumer.
+        /// - Parameter: A byte chunk.
+        /// - Returns: `true` if the byte chunk should be forwarded, `false` if this byte chunk is the terminating sequence.
+        let predicate: (ArraySlice<UInt8>) -> Bool
```

Comment:
Would it be possible to also remove the `predicate` property from the iterator type? I believe it should be enough for it to be owned by the state machine.

So it'd stay in the initializer of the iterator, but wouldn't be persisted as a property on the iterator, only forwarded to the state machine's iterator.

Reply:
Oh, of course. Done!